.gitlab-ci.yml

```diff
@@ -0,0 +1,152 @@
+variables:
+  PROJECT_ALLOC_NAME: ${CI_PROJECT_NAME}_ci_${CI_PIPELINE_ID}
+  BUILD_ROOT: ${CI_PROJECT_DIR}
+  # Should make it so we aren't doing fresh clones of GEOSX all the time
+  GIT_STRATEGY: fetch
+  # Manually update submodules
+  GIT_SUBMODULE_STRATEGY: none
+
+stages:
+  - build
+  - nightly
+
+####
+# Template
+.build_script:
+  script:
+    # Change submodule paths to absolute github paths (default is relative to GEOSX gitlab path)
+    - git submodule set-url src/coreComponents/LvArray https://github.com/GEOSX/LvArray.git
+    - git submodule set-url src/cmake/blt https://github.com/LLNL/blt.git
+    - git submodule set-url src/coreComponents/constitutive/PVTPackage https://github.com/GEOSX/PVTPackage.git
+    - git submodule set-url integratedTests git@github.com:GEOSX/integratedTests.git
+    - git submodule set-url src/coreComponents/mesh/PAMELA https://github.com/GEOSX/PAMELA.git
+    - git submodule set-url src/coreComponents/fileIO/coupling/hdf5_interface https://github.com/GEOSX/hdf5_interface.git
+
+    # Update submodules
+    - git submodule update --init --recursive src/cmake/blt
+    - git submodule update --init --recursive src/coreComponents/LvArray
+    - git submodule update --init --recursive src/coreComponents/constitutive/PVTPackage
+    - git submodule update --init --recursive integratedTests
+    - git submodule update --init src/coreComponents/mesh/PAMELA
+    - git submodule update --init --recursive src/coreComponents/fileIO/coupling/hdf5_interface
+
+    - mkdir ${SYSTEM}
+
+    # CONFIGURE
+    - echo "~~~~~~~~~~ START - configure ~~~~~~~~~~~"
+    - config_code=0
+    - python scripts/config-build.py -hc host-configs/LLNL/${HOST_CONFIG} -bt Release -bp build -DENABLE_DOXYGEN=OFF 2>&1 | tee ${SYSTEM}/${SYSTEM}_configure.log || config_code=$?
+    - if [[ $config_code -ne 0 ]]; then mv ${SYSTEM}/${SYSTEM}_configure.log ${SYSTEM}/${SYSTEM}_configure_FAILURE.log; else mv ${SYSTEM}/${SYSTEM}_configure.log ${SYSTEM}/${SYSTEM}_configure_SUCCESS.log; fi
+    - echo "~~~~~~~~~~ END - configure ~~~~~~~~~~~~~"
+
+    # BUILD
+    - echo "~~~~~~~~~~ START - build ~~~~~~~~~~~"
+    - build_code=0
+    - ${ALLOC_COMMAND} make VERBOSE=1 -C build -j ${NPROC} 2>&1 | tee ${SYSTEM}/${SYSTEM}_build.log || build_code=$?
```

Comment:
Where does this `NPROC` env variable comes from? Will it always be defined?
Or you could use `nproc` instead?

Reply:
`NPROC` can be defined through the Gitlab GUI, default will be undefined so will be `make -j`.

Reply:
I'm fine this but `make -j` can stuck the slave since `make` will run as many targets as it can, with no consideration on the number of processors.
You can consider something like `${NPROC:-$(nproc)}` or `${NPROC:-4}` which is surely safer.